src/Microsoft.IdentityModel.Tokens/SymmetricSignatureProvider.cs

```diff
@@ -377,7 +382,17 @@ internal bool Verify(byte[] input, int inputOffset, int inputLength, byte[] sign
             try
             {
                 keyedHashAlgorithm = GetKeyedHashAlgorithm(GetKeyBytes(Key), Algorithm);
-                return Utility.AreEqual(signature, keyedHashAlgorithm.ComputeHash(input, inputOffset, inputLength), signatureLength);
+
+                scoped Span<byte> hash;
+#if NET6_0_OR_GREATER
+                hash = stackalloc byte[keyedHashAlgorithm.HashSize / 8]; // only known algorithms are used, all of which have a small enough hash size to stackalloc
+                keyedHashAlgorithm.TryComputeHash(input.AsSpan(inputOffset, inputLength), hash, out int bytesWritten);
+                Debug.Assert(bytesWritten == hash.Length);
+#else
+                hash = keyedHashAlgorithm.ComputeHash(input, inputOffset, inputLength).AsSpan();
+#endif
+
+                return signature.AsSpan().StartsWith(hash.Slice(0, signatureLength));
```

Comment:
What about the timing concerns as outlined in Utility.AreEqual?

Reply:
I'm suspect of those timing concerns as outlined in AreEqual.  But even if we believe in them, I don't see how they're relevant to this particular consumption.  I'd be happy for someone to explain why I'm wrong about that, though :)

Reply:
@stephentoub where is the signature comparison performed?
Timing attacks are a real thing, we need to have a constant time value with comparisons.

Reply:
> where is the signature comparison performed?

The line this comment is on.

> Timing attacks are a real thing, we need to have a constant time value with comparisons.

I understand they're a real thing.  I'd be interested in understanding how one could manifest in this particular case.  But regardless I can switch this line back to using a variant of AreEqual modified to accept spans.

Reply:
> Maybe you can use https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.cryptographicoperations.fixedtimeequals?view=net-7.0 ?

It's not available on older TFMs.